### PR TITLE
Random generators for unif discrete and continuous

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: docs/reference/functions/proba/uniform_continuous_distribution.html [hunk 1]
```diff
@@ -8,20 +8,20 @@
  -->
 <meta charset="utf-8" lang="en">
                        **Expressive Vector Engine**
-                           `eve::uniform_real_distribution`
+                           `eve::uniform_discrete_distribution`
```
Comment (jfalcou) [inline docs/reference/functions/proba/uniform_continuous_distribution.html:11]: discrete or continuous ? the file says continuous

File: include/eve/module/real/proba/function/regular/generic/uniform_discrete_distribution.hpp [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+  {
+    //////////////////////////////////////////////////////
+    /// cdf discrete
+    template<typename T, typename U, floating_value V, typename I = V>
+    EVE_FORCEINLINE  auto cdf_(EVE_SUPPORTS(cpu_)
+                              , uniform_discrete_distribution<T, U, I> const & d
+                              , V const &x ) noexcept
+     requires compatible_values<V, typename uniform_discrete_distribution<T, U, I>::value_type>
+     {
+      if constexpr(value<T> && value<U>)
```
Comment (jfalcou) [inline include/eve/module/real/proba/function/regular/generic/uniform_discrete_distribution.hpp:142]: I don't understand the logic. T and U are of course value, when are we supposed to go the else ?

File: include/eve/module/real/proba/function/regular/generic/uniform_discrete_distribution.hpp [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    //////////////////////////////////////////////////////
+    /// pmf discrete
+    template<typename T, typename U, floating_value V
+             , typename I = T>
+    EVE_FORCEINLINE  auto pmf_(EVE_SUPPORTS(cpu_)
+                                 , uniform_discrete_distribution<T, U, I> const & d
+                              , V const &x ) noexcept
+    requires compatible_values<V, typename uniform_discrete_distribution<T, U, I>::value_type>
+    {
+      if constexpr(value<T> && value<U>)
```
Comment (jfalcou) [inline include/eve/module/real/proba/function/regular/generic/uniform_discrete_distribution.hpp:160]: Again same question

File: include/eve/module/real/proba/function/regular/generic/uniform_discrete_distribution.hpp [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+    //////////////////////////////////////////////////////
+    /// mgf discrete
+    template<typename T, typename U, floating_value V
+             , typename I = T>
+    EVE_FORCEINLINE  auto mgf_(EVE_SUPPORTS(cpu_)
+                                 , uniform_discrete_distribution<T, U, I> const & d
+                                 , V const &x ) noexcept
+    requires compatible_values<V, typename uniform_discrete_distribution<T, U, I>::value_type>
+    {
+      if constexpr(value<T> && value<U>)
```
Comment (jfalcou) [inline include/eve/module/real/proba/function/regular/generic/uniform_discrete_distribution.hpp:175]: Same question about value

File: include/eve/module/real/proba/function/regular/generic/uniform_discrete_distribution.hpp [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+        return inc(exp(x))*half(as(x));
+    }
+
+    //////////////////////////////////////////////////////
+    /// mean  discrete
+    template<typename T, typename U,  typename I = T>
+    EVE_FORCEINLINE  auto mean_(EVE_SUPPORTS(cpu_)
+                                  , uniform_discrete_distribution<T,U,I> const & d) noexcept
+    {
+      if constexpr (value<T> && value<U>)
```
Comment (jfalcou) [inline include/eve/module/real/proba/function/regular/generic/uniform_discrete_distribution.hpp:187]: Same question about value

File: test/unit/module/real/proba/uniform_continuous_distribution/attributes/uniform_continuous_distribution.hpp [hunk 6]
```diff
@@ -25,15 +25,15 @@ TTS_CASE_TPL("Check eve::uniform_real_distribution behavior", EVE_TYPE)
...
 
-  TTS_ULP_EQUAL(eve::median(d)               , T(0)  , 0);
-  TTS_ULP_EQUAL(eve::mean(d)                 , T(0.5)  , 0);
-  TTS_ULP_EQUAL(eve::var(d)                  , T(1/12.0)  , 0);
-  TTS_ULP_EQUAL(eve::stdev(d)                , T(2.886751345948129e-01)  , 1);
-  TTS_ULP_EQUAL(eve::cdf(d, T(0.0))          , T(0)   , 1.0);
-  TTS_ULP_EQUAL(eve::pdf(d, T(1.0))          , T(1)   , 1.0);
-}
+//   TTS_ULP_EQUAL(eve::median(d)               , T(0)  , 0);
+//   TTS_ULP_EQUAL(eve::mean(d)                 , T(0.5)  , 0);
```
Comment (jfalcou) [inline test/unit/module/real/proba/uniform_continuous_distribution/attributes/uniform_continuous_distribution.hpp:34]: Remove, don't comment

File: test/unit/module/real/proba/uniform_discrete_distribution/attributes/uniform_discrete_distribution.hpp [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  TTS_ULP_EQUAL(eve::median(d)               , T(1.0)  , 0);
+  TTS_ULP_EQUAL(eve::mean(d)                 , T(1.5)  , 0);
+  TTS_ULP_EQUAL(eve::var(d)                  , T(0.25)  , 0);
+  TTS_ULP_EQUAL(eve::stdev(d)                , T(0.5)  , 1);
+  TTS_ULP_EQUAL(eve::cdf(d, T(0.0))          , T(0)   , 1.0);
+  TTS_ULP_EQUAL(eve::cdf(d, T(1.5))          , T(0.5) , 1.0);
+  TTS_ULP_EQUAL(eve::pmf(d, T(1.0))          , T(0.5)   , 1.0);
+}
+
+// TTS_CASE_TPL("Check eve::uniform_discrete_distribution behavior", EVE_TYPE)
```
Comment (jfalcou) [inline test/unit/module/real/proba/uniform_discrete_distribution/attributes/uniform_discrete_distribution.hpp:28]: Remove, don't comment

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/module/real/proba/function/regular/generic/uniform_discrete_distribution.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/real/proba/function/regular/generic/uniform_discrete_distribution.hpp:142]: no they can be callable one or zero

File: include/eve/module/real/proba/function/regular/generic/uniform_discrete_distribution.hpp (hunk 3 above)
Comment (jtlap) [reply include/eve/module/real/proba/function/regular/generic/uniform_discrete_distribution.hpp:160]: same answer

**Review by jfalcou — APPROVED** [review]

**Pull request merged**